### ISSUE TITLE
Blog Posts of Jupyter Notebooks

### ISSUE THREAD

**Issue by oke-aditya** [madewithml/hacktoberfest#16]
## Maintenance Issue

Documentation / Display.

A few thoughts that I had were how to make this more accessible.
We have mkdocs that give us an overview of stuff right now. It is cool, it has material design and creates stuff.

I am thinking of [fastpages](https://github.com/fastai/fastpages). This supports Jupyter notebooks and helps in getting our notebooks converted to blogs out of the box.

It has CI action as well which makes it very simple to deploy. We can generate static pages over `gh-pages` and redirect with custom `CNAME` which can easily then be going to made with ml. We won't have to worry much about `load-balancing` etc I believe though.

This is something in line to LearnOpenCV (biggest CV blog ever). I am not sure how to add this up. Maybe someone can look into this. 

Please let me know thoughts and alternatives.

@ayulockin  @GokuMohandas

**Comment by GokuMohandas** [comment]
Definitely love the idea of using fastpages for this and yup it'll be as simple as making these [few additions](https://github.com/fastai/fastpages/blob/master/_fastpages_docs/_setup_pr_template.md#optional-using-a-custom-domain) to make it work for a custom domain.

**Comment by oke-aditya** [comment]
If this works. I will remove mkdocs (I don't think it will be advisable to have 2 docmentations)

**Comment by GokuMohandas** [comment]
They both seem quite good (and easy to use) to be honest and it may come down to a stylistic preference.

Fastpages: https://fastpages.fast.ai/actions/markdown/2020/03/06/fastpages-actions.html
MKDocs (with material theme): https://koaning.github.io/clumper/guides/verbs.html

Main differences seem to be high level side panel for entire docs (not just for the post), tabs, etc. I see most people using fastpages for blogposts and mkdocs and other options for anything else.

**Comment by oke-aditya** [comment]
I have built mkdocs with the material theme 😄  I have used [this](https://squidfunk.github.io/mkdocs-material/)

It's already live [here](https://madewithml.github.io/hacktoberfest/)

I see fastpages better as the left panel might get crammed up after many jupyter notebooks. So preferrable fastpages.

**Comment by GokuMohandas** [comment]
Good point that can definitely get cluttered. Alright fastpages it is!

**Comment by oke-aditya** [comment]
It would also be nice, if we can generate fastpages like netlify preview on every PR.

**Comment by oke-aditya** [comment]
Over to you 😄  @neomatrix369

**Comment by neomatrix369** [comment]
> Over to you 😄 @neomatrix369

Ok will look at it asap! It's quite a broad topic, the more I look into it

**Comment by neomatrix369** [comment]
> Over to you 😄 @neomatrix369

Just to be clear @oke-aditya which things from this PR would you like me to address, as there a couple of things for Goku to action as well.

Basically we wish to setup fastpages to be able to render our blogs/notebooks into them and also want to connect it to CI/CD so its automated right? I will need to look a bit closer this week to get this rolling.

**Comment by oke-aditya** [comment]
Yes, let me summarize.

- [ ] Setup fastpages. It will go to www.guides.madewithml.com url. We have this domain.
- [ ] Have netlify build preview triggered for fastapges.
On every PR we should show the netlify preview of fastpages blog that would be generated.
This would be triggered on every Pull request commit. 
Netlify offers this for free 😄 

For these 2 helping steps (which I can do) are.

- [ ] Clean mkdocs (we had material mkdocs which won't work well)
- [ ] Clean gh-pages branch
- [ ] CNAME file to redirect people.

**Comment by neomatrix369** [comment]
Perfect thanks!

**Comment by oke-aditya** [comment]
Yes you can create GitHub actions, they will be triggered on Pull request. But let us know if they don't
Anyways I fell you should get more access 😄 

cc @GokuMohandas

**Comment by oke-aditya** [comment]
Hey, @neomatrix369  any updates? Most probably with Fastpages we can launch this 😃 

Goku and I discussed about netlify and we evaluated that it is still viable to use it. Despite having build limitations

**Comment by neomatrix369** [comment]
> Hey, @neomatrix369 any updates? Most probably with Fastpages we can launch this 😃
> 
> Goku and I discussed about netlify and we evaluated that it is still viable to use it. Despite having build limitations

Sorry @oke-aditya being a bit slow getting to this - its right on the top of my list

**Comment by oke-aditya** [comment]
No issue @neomatrix369 lets go slow and steady. We should setup things so tha it should be easy to maintain.

**Comment by neomatrix369** [comment]
> Yes, let me summarize.
> 
> * [ ]  Setup fastpages. It will go to [www.guides.madewithml.com](http://www.guides.madewithml.com) url. We have this domain.
> * [ ]  Have netlify build preview triggered for fastapges.
>   On every PR we should show the netlify preview of fastpages blog that would be generated.
>   This would be triggered on every Pull request commit.
>   Netlify offers this for free 😄
> 
> For these helping steps (which I can do) are.
> 
> * [x]  Clean mkdocs (we had material mkdocs which won't work well)
> * [x]  Clean gh-pages branch
> * [x]  CNAME file to redirect people.
> 
> Fastpages has a github action (CI) that makes setting it up simple.
> I'm not sure about netlify but I guess it should be possible.
> 
> Let me and Goku know what else can be done 😄
> 
> Fastpages is necessary, we can add netlify later. Or even someone else too can look into it (if needed).

@oke-aditya do we have a sample/or real PR with a notebook in it so that I can start generating the FastPage first before we do the next interaction with Netlify? Also is the target URL http://www.guides.madewithml.com or http://guides.madewithml.com. 

What's unclear to me (and maybe this may get clear later on) - how do we layout the blogs on the http://www.guides.madewithml.com URL? By categories? Tabs? What would the layout look like?

**Comment by oke-aditya** [comment]
Hi @neomatrix369 . A lot of questions here. Let me answer to best of my knowledge.

1. We can take [this](https://github.com/madewithml/guides/pull/17) PR and it's notebook as a starting point. The PR looks good to me and probably nice to add. Feel free to make changes if necessary.

2. Which URL -> I will leave to @GokuMohandas I am not sure. But I think it will be `https://`

3. For layout, posts etc. I think we should discuss over a meet. 
Wireframing and some prototyping is quite tricky. This might be trial and error and maybe few people should think over this.

**Comment by neomatrix369** [comment]
> Hi @neomatrix369 . A lot of questions here. Let me answer to best of my knowledge.
> 
> 1. We can take [this](https://github.com/madewithml/guides/pull/17) PR and it's notebook as a starting point. The PR looks good to me and probably nice to add. Feel free to make changes if necessary.
> 2. Which URL -> I will leave to @GokuMohandas I am not sure. But I think it will be `https://`
> 3. For layout, posts etc. I think we should discuss over a meet.
>    Wireframing and some prototyping is quite tricky. This might be trial and error and maybe few people should think over this.

Thanks for the response, Ill get on with what I have and if I get stuck somewhere I will let you know.
